lib/cocoapods/installer/project_cache/target_cache_key.rb

```diff
@@ -51,10 +51,13 @@ def key_difference(other)
               return :project if (other.key_hash.keys - key_hash.keys).any?
               return :project if other.key_hash['CHECKSUM'] != key_hash['CHECKSUM']
               return :project if other.key_hash['SPECS'] != key_hash['SPECS']
-              return :project if other.key_hash['FILES'] != key_hash['FILES']
               return :project if other.key_hash['PROJECT_NAME'] != key_hash['PROJECT_NAME']
             end
 
+            this_files = key_hash['FILES']
+            other_files = other.key_hash['FILES']
+            return :project if this_files != other_files
```

Comment:
Why did this get pulled out?

Reply:
Because we need to compare file lists for all targets now,

Reply:
Cool